app/add-details/page.tsx

```diff
@@ -0,0 +1,147 @@
+
+  const plants: Plant[] = [
+    {
+      id: '43c19f80-8205-4d03-b323-05c220550bf0',
+      plant_name: 'cabbage',
+      us_state: 'string',
+      harvest_season: 'SPRING',
+      water_frequency: 'string',
+      weeding_frequency: 'string',
+      indoors_start: 'string',
+      indoors_end: 'string',
+      outdoors_start: 'string',
+      outdoors_end: 'string',
+      transplant_start: 'string',
+      transplant_end: 'string',
+      harvest_start: 'string',
+      harvest_end: 'string',
+      beginner_friendly: true,
+      plant_tips: 'string',
+      img: 'string',
+      difficulty_level: 'HARD',
+      sunlight_min_hours: 1,
+      sunlight_max_hours: 1,
+    },
+    {
+      id: '43c19f80-8205-4d03-b323-05c220550bf0',
+      plant_name: 'tomato',
+      us_state: 'string',
+      harvest_season: 'SPRING',
+      water_frequency: 'string',
+      weeding_frequency: 'string',
+      indoors_start: 'string',
+      indoors_end: 'string',
+      outdoors_start: 'string',
+      outdoors_end: 'string',
+      transplant_start: 'string',
+      transplant_end: 'string',
+      harvest_start: 'string',
+      harvest_end: 'string',
+      beginner_friendly: true,
+      plant_tips: 'string',
+      img: 'string',
+      difficulty_level: 'HARD',
+      sunlight_min_hours: 1,
+      sunlight_max_hours: 1,
+    },
+  ];
+  const user_id: UUID = 'e72af66d-7aae-45f6-935a-187197749d9f';
+  const getDefaultDate = () => new Date().toISOString().substring(0, 10);
+
+  function move(steps: number) {
+    // Navigate between plants and save input data
+    const currentDetail = details[currentIndex - 1];
+    if (
+      (!currentDetail || !currentDetail.date_added) &&
+      currentIndex <= plants.length
+    ) {
+      updateInput('date_added', getDefaultDate());
+    }
+    if (
+      steps > 0 &&
+      (!currentDetail?.planting_type ||
+        currentDetail?.planting_type == 'SELECT')
+    ) {
+      alert('Please select a valid planting type and date before proceeding.');
+      return;
+    }
+    if (
+      steps !== 0 &&
+      currentIndex + steps > 0 &&
+      currentIndex + steps <= plants.length + 1
+    ) {
+      if (steps > 0) {
+        setIsDisabled(true);
+      }
+      setCurrentIndex(prevIndex => prevIndex + steps);
+    }
+  }
+
+  function updateInput(field: string, value: string) {
+    // Update the specific field of the current plant's details
+    if (field == 'planting_type') {
+      let shouldBeDisabled = isDisabled;
+      if (value === 'SELECT') {
+        shouldBeDisabled = true;
+      } else {
+        shouldBeDisabled = false;
+      }
+      setIsDisabled(shouldBeDisabled);
+    }
+    const updatedDetails = [...details];
+    updatedDetails[currentIndex - 1] = {
+      ...updatedDetails[currentIndex - 1],
+      [field]: value,
+      plant_id: plants[currentIndex - 1].id,
+    };
+    setDetails(updatedDetails);
+  }
+
+  function updateDB() {
+    updateUserPlants(user_id, details);
```

Comment:
This should be async, so we should modify this to: 
```
async function updateDB() {
    await updateUserPlants(user_id, details);
    router.push('/view-plants');
}
```
Awaiting ensures that updateUserPlants is completed before rerouting to /view-plants, so that the changes made in the Add Plant Details flow are reflected in  /view-plants.